examples/integer-localizations.jl

```diff
@@ -0,0 +1,222 @@
+
+@Markdown.doc """
+FmpzPowersOfElement
+
+The multiplicative set given by the powers of some element.
+"""
+mutable struct FmpzPowersOfElement <: AbsMultSet{FlintIntegerRing, fmpz}
+  d::Vector{fmpz} # a vector of admissible denominators. 
+
+  function FmpzPowersOfElement(denominators::Vector{fmpz})
+    return new(denominators)
+  end
+end
+
+### required getter functions
+ambient_ring(S::FmpzPowersOfElement) = ZZ
+
+### required functionality
+function Base.in(a::fmpz, S::FmpzPowersOfElement) 
+  # check whether ∃ c ∈ ℤ, k ∈ ℕ₀: c ⋅ a = (∏ᵢ dᵢ)ᵏ, where dᵢ are the admissible denominators in S.
+  d = prod(denominators(S))
+  g = gcd(a, d)
+  while(!isone(g))
+    a = divexact(a, g)
+    g = gcd(a, d)
+  end
+  return isone(a)
+end
+
+### additional constructors
+FmpzPowersOfElement(d::fmpz) = FmpzPowersOfElement([d])
+FmpzPowersOfElement(d::Oscar.IntegerUnion) = FmpzPowersOfElement(ZZ(d))
+FmpzPowersOfElement(d::Vector{T}) where {T<:Oscar.IntegerUnion} = FmpzPowersOfElement(ZZ.(d))
+
+### additional functionality
+denominators(S::FmpzPowersOfElement) = S.d::Vector{fmpz}
+Base.in(a::Oscar.IntegerUnion, S::FmpzPowersOfElement) = (ZZ(a) in S)
+
+
+#################################################################################
+# Complements of prime ideals in ℤ                                              #
+#################################################################################
+
+@Markdown.doc """
+FmpzComplementOfPrimeIdeal
+
+The multiplicatively closed set `S = ℤ ∖ ⟨p⟩` of integers outside a prime ideal `⟨p⟩`.
+"""
+mutable struct FmpzComplementOfPrimeIdeal <: AbsMultSet{FlintIntegerRing, fmpz}
+  # essential fields
+  p::fmpz
+
+  function FmpzComplementOfPrimeIdeal(p::fmpz)
+    isprime(p) || error("$(p) is not prime")
+    return new(p)
+  end
+end
+
+### required getter functions
+ambient_ring(S::FmpzComplementOfPrimeIdeal) = ZZ
+
+### required functionality
+function Base.in(b::fmpz, S::FmpzComplementOfPrimeIdeal)
+  return !(divides(b, S.p)[1])
+end
+
+### additional constructors
+FmpzComplementOfPrimeIdeal(i::Integer) = FmpzComplementOfPrimeIdeal(ZZ(i))
+
+### additional functionality
+Base.in(a::Oscar.IntegerUnion, S::FmpzComplementOfPrimeIdeal) = (ZZ(a) in S)
+
+
+#################################################################################
+# The complement of the zero ideal in ℤ                                         #
+#################################################################################
+
+@Markdown.doc """
+FmpzComplementOfZeroIdeal 
+
+The complement of the zero ideal in `ℤ`.
+"""
+mutable struct FmpzComplementOfZeroIdeal <: AbsMultSet{FlintIntegerRing, fmpz}
+  function FmpzComplementOfZeroIdeal() 
+    return new{}()
+  end
+end
+
+### required getter functions
+ambient_ring(S::FmpzComplementOfZeroIdeal) = ZZ
+
+### required functionality
+function Base.in(b::fmpz, S::FmpzComplementOfZeroIdeal) 
+  return !(b == zero(ZZ))
+end
```

Comment:
```suggestion
Base.in(b::fmpz, S::FmpzComplementOfZeroIdeal) = !iszero(b)
```

Reply:
Ok, this is weird. I thought, I had applied such changes everywhere. But now it seems, I have forgotten this file. Sorry.

Reply:
This should now finally be taken care of.